src/main/java/dev/streamx/sling/connector/impl/StreamxPublicationServiceImpl.java

```diff
@@ -68,12 +82,42 @@ public boolean isEnabled() {
 
   @Override
   public void publish(List<String> pathsToPublish) throws StreamxPublicationException {
-    handlePublication(PublicationAction.PUBLISH, pathsToPublish);
+    submitIngestionTriggerJob(PublicationAction.PUBLISH, pathsToPublish);
   }
 
   @Override
   public void unpublish(List<String> pathsToUnpublish) throws StreamxPublicationException {
-    handlePublication(PublicationAction.UNPUBLISH, pathsToUnpublish);
+    submitIngestionTriggerJob(PublicationAction.UNPUBLISH, pathsToUnpublish);
+  }
+
+  private void submitIngestionTriggerJob(
+      PublicationAction ingestionAction, Collection<String> pathsToIngest
+  ) {
+    List<SlingUri> slingUris = pathsToIngest.stream()
```

Comment:
what is the purpose of the SlingIUri usage? we get strings, we map to SlingUri, then map to string to save in job properties, then read from job as strings, then map to SlingUri and again map to string to pass to handlers.